### PR TITLE
Add directory creation logic utility to Fs.h

### PR DESCRIPTION
**Pull request by tanjialiang**
Add this commonly used util to Fs.h. Directories will be created 
recursively from the newly added functionality.

### CODE REVIEW

File: velox/vector/VectorSaver.h [hunk 1]
```diff
@@ -66,4 +66,12 @@ std::string restoreStringFromFile(const char* FOLLY_NONNULL filePath);
 std::optional<std::string> generateFilePath(
     const char* FOLLY_NONNULL basePath,
     const char* FOLLY_NONNULL prefix);
+
+/// Generates a file directory specified by 'dirPath'. The generation will be
+/// recursive. Non-exist parent directories will also be created. Returns true
+/// if creation is successful, false otherwise. Error message will be printed if
+/// creation is unsuccessful, but already created directories will not be
+/// removed.
+bool generateFileDirectory(const std::string& dirPath);
```
Comment (bikramSingh91) [inline velox/vector/VectorSaver.h:75]: just looked up the cpp filesystem interface, turns out you can use std::filesystem::create_directories() that recursively generates directories.

Comment (tanjialiang) [reply velox/vector/VectorSaver.h:75]: Turns out so! Thanks for the pointer. Let me make the changes

Comment (mbasmanova) [reply velox/vector/VectorSaver.h:75]: FYI, see velox/common/base/Fs.h

**Review by bikramSingh91 — APPROVED** [review]

**Review by mbasmanova — COMMENTED** [review]
@tanjialiang Please, update the PR title. It doesn't match the code.

**Review by aditi-pandit — APPROVED** [review]

**Pull request closed without merging**